modules/tracing/mule-tracer-exporter-configuration-impl/src/main/java/org/mule/runtime/tracer/exporter/api/config/FileSpanExporterConfiguration.java

```diff
@@ -31,43 +41,81 @@
 public class FileSpanExporterConfiguration implements SpanExporterConfiguration {
 
+  @Inject
+  MuleContext muleContext;
+
   private static final String PROPERTIES_FILE_NAME = "tracer-exporter.conf";
 
-  private final ConfigurationPropertiesResolver propertyResolver;
-  private final Properties properties;
+  private static final Logger LOGGER = getLogger(FileSpanExporterConfiguration.class);
 
-  public FileSpanExporterConfiguration() {
-    properties = getSpanExporterConfiguration();
-    propertyResolver =
-        new DefaultConfigurationPropertiesResolver(empty(),
-                                                   new SystemPropertiesConfigurationProvider());
-  }
+  private ConfigurationPropertiesResolver propertyResolver;
+  private Properties properties;
+  private ClassLoaderResourceProvider resourceProvider;
+  private boolean propertiesInitialised;
 
   @Override
   public String getStringValue(String key) {
+    if (!propertiesInitialised) {
+      initialiseProperties();
+      propertiesInitialised = true;
+    }
+
     String value = properties.getProperty(key);
 
     if (value != null) {
-      return propertyResolver.apply(properties.getProperty(key));
+      // Verify if it is a system property.
+      value = propertyResolver.apply(value);
+
+      if (isAValueCorrespondingToAPath(key)) {
+        // Obtain absolute path and return it if possible.
+        String absolutePath = getAbsolutePath(value);
+
+        if (absolutePath != null) {
+          return absolutePath;
+        }
+        // Otherwise, return the non-resolved value so that the error message makes sense.
+      }
+
+      return value;
+    }
+
+    return null;
+  }
+
+  private String getAbsolutePath(String value) {
+    Path path = Paths.get(value);
+
+    try {
+      if (!path.isAbsolute()) {
+        URL url = getExecutionClassLoader(muleContext).getResource(value);
+
+        if (url != null) {
+          return new File(url.toURI()).getAbsolutePath();
+        }
+      }
+    } catch (URISyntaxException e) {
+      return value;
     }
 
     return null;
   }
 
+  private boolean isAValueCorrespondingToAPath(String key) {
+    return key.equals(MULE_OPEN_TELEMETRY_EXPORTER_CA_FILE_LOCATION) ||
+        key.equals(MULE_OPEN_TELEMETRY_EXPORTER_KEY_FILE_LOCATION);
+  }
+
   private Properties getSpanExporterConfiguration() {
     try {
-      InputStream is = getResourceAsStream(getConfFolder() + FileSystems.getDefault().getSeparator() + getPropertiesFileName(),
-                                           FileSpanExporterConfiguration.class);
+      // This will verify first in the app and then in the conf folder.
+      InputStream is = resourceProvider.getResourceAsStream(getPropertiesFileName());
       return loadProperties(is);
-    } catch (IOException e) {
+    } catch (MuleRuntimeException | IOException e) {
+      LOGGER.info("No tracer exporter config found in app. Loading it from the conf directory.");
       return getProperties();
```

Comment:
This checks the conf directory?

Reply:
yes, the other verification is not needed.  I didn't changed the logger info message. I've changed it now.